src/Propel/Generator/Behavior/Versionable/VersionableBehaviorObjectBuilderModifier.php

```diff
@@ -246,22 +251,49 @@ public function isVersioningNecessary(\$con = null)
     }
 ";
         }
-        $plural = true;
+
         foreach ($this->behavior->getVersionableReferrers() as $fk) {
-            $fkGetter = $this->builder->getRefFKPhpNameAffix($fk, $plural);
-            $script .= "
-    // to avoid infinite loops, emulate in save
-    \$this->alreadyInSave = true;
-    foreach (\$this->get{$fkGetter}(null, \$con) as \$relatedObject) {
-        if (\$relatedObject->isVersioningNecessary(\$con)) {
-            \$this->alreadyInSave = false;
 
+            if ($fk->isLocalPrimaryKey()) {
+
+                $fkGetter = $this->builder->getRefFKPhpNameAffix($fk);
+                $script .= "
+    if (\$this->single{$fkGetter}) {
+
+        // to avoid infinite loops, emulate in save
+        \$this->alreadyInSave = true;
+
+        if (\$this->single{$fkGetter}->isVersioningNecessary(\$con)) {
+
+            \$this->alreadyInSave = false;
```

Comment:
where exactly happens that infinite loop? You set `alreadyInSave` but never check it?